### PR TITLE
feat(replays): Add dead letter queue policy to storage stream-loader

### PR DESCRIPTION
**Pull request by cmanallen**
This pull adds a dead-letter-queue policy to the replays storage stream-loader.  I followed the instructions Rahul outlined here: https://github.com/getsentry/snuba/pull/2806#issuecomment-1155794695.

I'm under the impression that this is all I need but I'm new to the Snuba library.  Let me know if I missed something.

### CODE REVIEW

File: snuba/datasets/storages/replays.py [hunk 1]
```diff
@@ -58,7 +66,14 @@
     storage_set_key=StorageSetKey.REPLAYS,
 )
 
-# TODO: set up deadletter queue for bad messages.
+
+def produce_policy_creator() -> DeadLetterQueuePolicy:
+    """Produce all bad messages to dead-letter topic."""
+    return ProduceInvalidMessagePolicy(
+        KafkaProducer(build_kafka_producer_configuration(Topic.DEAD_LETTER_METRICS)),
+        KafkaTopic(Topic.DEAD_LETTER_METRICS.value),
```
Comment (fpacifici) [inline snuba/datasets/storages/replays.py:74]: You will need your own topic. That is the Topic we use for the metrics consumer.
Please add one in the topics file here https://github.com/getsentry/snuba/blob/master/snuba/utils/streams/topics.py#L28-L31

**Review by fpacifici — CHANGES_REQUESTED** [review]
Please register the missing topics. 
Once that is done you will need to ask the ops team to create the topic and add it to the production config. In order to achieve that, please contact the search and storage team in slack discuss-eng-sns.

**Comment by rahul-kumar-saini** [comment]
Echoing Filippo's comment, you should create a new dead-letter topic by filing an [ops request](https://getsentry.atlassian.net/secure/CreateIssue.jspa?issuetype=10301&pid=10303) for it. It should be created on the same cluster as the replay events topic. We have a simple convention going `snuba-dead-letter-XXX`.

It can then be added to the production config [here](https://github.com/getsentry/ops/blob/master/k8s/services/snuba/templates/prod_settings.py#L205-L239).
Probably just something like this:
```
  'snuba-dead-letter-replay-events': {
    'bootstrap.servers': KAFKA_SPECTO_BOOTSTRAP_SERVERS,
  },
```

File: snuba/datasets/storages/replays.py (hunk 1 above)
Comment (cmanallen) [reply snuba/datasets/storages/replays.py]: 👍

File: snuba/utils/streams/topics.py [hunk 2]
```diff
@@ -29,6 +29,7 @@ class Topic(Enum):
     DEAD_LETTER_METRICS = "snuba-dead-letter-metrics"
     DEAD_LETTER_SESSIONS = "snuba-dead-letter-sessions"
     DEAD_LETTER_GENERIC_METRICS = "snuba-dead-letter-generic-metrics"
+    DEAD_LETTER_REPLAYS = "snuba-dead-letter-replays"
```
Comment (fpacifici) [inline snuba/utils/streams/topics.py:32]: You will also need to add the topic to the config validation rules.
https://github.com/getsentry/snuba/blob/master/snuba/settings/validation.py#L21-L46

Comment (cmanallen) [reply snuba/utils/streams/topics.py:32]: @fpacifici Forgot to push after my last comment.  Should be good to go.

**Review by rahul-kumar-saini — APPROVED** [review]
LGTM pending https://github.com/getsentry/ops/pull/5018

**Comment by cmanallen** [comment]
@fpacifici Is this ready?

**Review by fpacifici — APPROVED** [review]

**Pull request merged**